applicationset/controllers/applicationset_controller.go

```diff
@@ -1597,10 +1597,23 @@ func getApplicationSetOwnsHandler() predicate.Funcs {
 }
 
 // shouldRequeueForApplicationSet determines when we need to requeue an applicationset
-func shouldRequeueForApplicationSet(appSetOld, appSetNew *argov1alpha1.ApplicationSet) bool {
+func shouldRequeueForApplicationSet(appSetOld, appSetNew *argov1alpha1.ApplicationSet, enableProgressiveSyncs bool) bool {
 	if appSetOld == nil || appSetNew == nil {
 		return false
 	}
+
+	// Requeue if any ApplicationStatus.Status changed for Progressive sync strategy
+	if enableProgressiveSyncs {
+		for _, newApplicationStatus := range appSetNew.Status.ApplicationStatus {
+			idx := findApplicationStatusIndex(appSetOld.Status.ApplicationStatus, newApplicationStatus.Application)
+			if idx != -1 {
+				if appSetOld.Status.ApplicationStatus[idx].Status != newApplicationStatus.Status {
+					return true
+				}
+			}
+		}
+	}
```

Comment:
```suggestion
	if enableProgressiveSyncs {
		if !cmp.Equal(appSetOld.Status.ApplicationStatus, appSetNew.Status.ApplicationStatus, cmpopts.EquateEmpty()) {
			return true
		}
	}
```
any reason not to use `cmp.Equal`?